### PR TITLE
update for azure/aws provider

### PR DESCRIPTION
**Pull request by ropatil010**
Hi Team, 

Kindly Please help to review the file for azure provider. 
Platform: Azure
Payload: 4.16.0-0.nightly-2024-06-14-130320
secret driver version: 4.16.0-202406131906

Tests execution:
rohitpatil@ropatil-mac secrets-store-csi-driver % bats -t test/bats/azure.bats 
1..19
ok 1 install azure provider
ok 2 create azure k8s secret
ok 3 deploy azure secretproviderclass crd
ok 4 CSI inline volume test with pod portability
ok 5 CSI inline volume test with pod portability - read azure kv secret from pod
ok 6 CSI inline volume test with pod portability - read azure kv key from pod
ok 7 CSI inline volume test with pod portability - unmount succeeds
ok 8 Sync with K8s secrets - create deployment
ok 9 Sync with K8s secrets - read secret from pod, read K8s secret, read env var, check secret ownerReferences with multiple owners
ok 10 Sync with K8s secrets - delete deployment, check owner ref updated, check secret deleted
ok 11 Test Namespaced scope SecretProviderClass - create deployment
ok 12 Test Namespaced scope SecretProviderClass - Sync with K8s secrets - read secret from pod, read K8s secret, read env var, check secret ownerReferences
ok 13 Test Namespaced scope SecretProviderClass - Sync with K8s secrets - delete deployment, check secret deleted
ok 14 Test Namespaced scope SecretProviderClass - Should fail when no secret provider class in same namespace
ok 15 deploy multiple azure secretproviderclass crd
ok 16 deploy pod with multiple secret provider class
ok 17 CSI inline volume test with multiple secret provider class
ok 18 Test auto rotation of mount contents and K8s secrets - Create deployment
ok 19 Test auto rotation of mount contents and K8s secrets

cc: @dobsonj @jsafrane @duanwei33

### CODE REVIEW

**Comment by ropatil010** [comment]
Solution2:
Preserve the resource group, keyvault with values saved so that it will be always same across different azure clusters. 
Then run tc so that it can cross verify the values written inside with types: key, secret which will be always same
ex: KEYVAULT_KEY_VALUE(KEY_VALUE_CONTAINS)

/assign @dobsonj @jsafrane @duanwei33

**Comment by ropatil010** [comment]
Platform: AWS HCP kind
Provider: AWS-provider 
Payload: 4.16.0-0.nightly-2024-06-23-120416

Tests executed: 
rohitpatil@ropatil-mac secrets-store-csi-driver % bats -t test/bats/aws.bats 
1..9
ok 1 Install aws provider
ok 2 deploy aws secretproviderclass crd
ok 3 CSI inline volume test with pod portability
ok 4 CSI inline volume test with rotation - parameter store
ok 5 CSI inline volume test with rotation - secrets manager
ok 6 CSI inline volume test with pod portability - read ssm parameters from pod
ok 7 CSI inline volume test with pod portability - read secrets manager secrets from pod
ok 8 Sync with Kubernetes Secret
ok 9 CSI inline volume test with pod portability - unmount succeeds

/assign @xingxingxia

File: test/bats/aws.bats [hunk 1]
```diff
@@ -4,14 +4,19 @@ load helpers
...
-PROVIDER_YAML=https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml
-NAMESPACE=kube-system
+# PROVIDER_YAML=https://raw.githubusercontent.com/aws/secrets-store-csi-driver-provider-aws/main/deployment/aws-provider-installer.yaml
 POD_NAME=basic-test-mount
-export REGION=${REGION:-us-west-2}
-
-export ACCOUNT_NUMBER=$(aws --region $REGION  sts get-caller-identity --query Account --output text)
 BATS_TEST_DIR=test/bats/tests/aws
 
+export REGION=${REGION:-us-west-2}
```
Comment (Phaow) [inline test/bats/aws.bats:11]: From my understanding, we should get the region from the test cluster, rigit?
```suggestion
export REGION=$(oc get no -o=jsonpath='{.items[0].metadata.labels.topology\.kubernetes\.io/region}')
```

**Comment by ropatil010** [comment]
Based on earlier Discussion:  https://redhat-internal.slack.com/archives/C02221SB07R/p1717569901233769?thread_ts=1717568816.927779&cid=C02221SB07R initiated this PR and currently holding it based on latest discussion: https://redhat-internal.slack.com/archives/GK0DA0JR5/p1719423024345889 
/hold

**Pull request closed without merging**